### ISSUE TITLE
[Bug]: Tab scroll arrow and gap between default line tab

### ISSUE THREAD

**Issue by divya-281** [carbon-design-system/carbon#17648]
### Package

@carbon/react

### Package version

1.67.0

### Description

https://github.com/user-attachments/assets/4e34c76a-0a61-48f3-82a9-1e5ff7fe1ad1

### Reproduction/example

Can be seen on [story book](https://react.carbondesignsystem.com/?path=/story/components-tabs--default)

### Steps to reproduce

Can be seen on [story book](https://react.carbondesignsystem.com/?path=/story/components-tabs--default)

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by divya-281** [comment]
Hi @tay1orjones @alisonjoseph Can you assign this issue to me?

**Comment by divya-281** [comment]
Hi @aagonzales This issue was created for the issues you have mentioned in this [PR](https://github.com/carbon-design-system/carbon/pull/16522).  For the below issue:

> The arrow button seems to be a bit delayed in disappearing when dragging the browser back out. Not sure if others are seeing this too.

does this issue still exist? This is what I see:

https://github.com/user-attachments/assets/f4436e25-0d8e-43da-901e-f40e5f95be18

**Comment by aagonzales** [comment]
Yeah I still see a slight delay on it. Its not the biggest bug but it would be nice if the arrow disappeared as soon as the last tab hit full width. 

![Image](https://github.com/user-attachments/assets/a907b24c-a0cc-4c51-81ec-1a804a8f1dd0)

**Comment by divya-281** [comment]
@aagonzales The 1px gap between the tabs only disappears when the tab become scrollable. I don't know if we should have a gap between the tabs while scrolling, is this change needed?
